### DISCUSSION TITLE
Building a Go SDK Implementation (Help Needed)

### DISCUSSION THREAD

**Discussion by ezynda3** [modelcontextprotocol/.github#19] [Show and tell]
I've scaffolded a basic SDK implementation in Go with the help of the specification docs and Claude Sonnet. Would have taken forever otherwise lol.

I have a basic client implemented with stdio and sse transports. Would love to get this to parity with the JS and Python SDKs. I'm ok with Go but not an expert by any means. Happy to have some help or at least have people check out the lib and note what's missing or can be fixed.

Cheers!

https://github.com/mark3labs/mcp-go

**Comment by ezynda3** [comment]
I heavily refactored a bunch of stuff but now I think it's looking a bit better. So far we can
- Create a stdio client
- Create a stdio server

Technically it's possible to wrap a default server in any transport but the library only supports wrapping in stdio for now

**Comment by ezynda3** [comment]
We now have SSE Client/Server support!

**Comment by mohanz** [comment]
nice!. I made a small rust version and linked your repo
https://github.com/AntigmaLabs/mcp-sdk
